coriolis/providers/backup_writers.py

```diff
@@ -213,29 +220,98 @@ def _send_msg(self, data):
                 raise exception.CoriolisException(
                     "write_data exited with error code %r (%s)" % (
                         ret_val, _WRITER_ERR_MAP.get(int(ret_val))))
-        self._msg_id += 1
+
         self._stdin.write(data)
         self._stdin.flush()
         self._stdout.read(4)
 
     def _open(self):
         self._exec_helper_cmd()
+        self._sender_evt = eventlet.spawn(
+            self._sender)
+        self._encoder_evt = [
+            eventlet.spawn(self._encoder),
```

Comment:
NIT: I'd suggest we add a configuration option for this instead of hardcoding it to 3

Reply:
done